reconcile/utils/openshift_resource.py

```diff
@@ -443,17 +443,23 @@ def initialize_resource_type(self, cluster, namespace, resource_type):
         self._clusters[cluster].setdefault(namespace, {})
         self._clusters[cluster][namespace].setdefault(resource_type, {
             'current': {},
-            'desired': {}
+            'desired': {},
+            'use_admin_token': {}
         })
 
-    def add_desired(self, cluster, namespace, resource_type, name, value):
+    def add_desired(self, cluster, namespace, resource_type, name, value,
+                    privileged=False):
         with self._lock:
             desired = \
                 (self._clusters[cluster][namespace][resource_type]
                     ['desired'])
             if name in desired:
                 raise ResourceKeyExistsError(name)
             desired[name] = value
+            admin_token_usage = \
+                (self._clusters[cluster][namespace][resource_type]
+                    ['use_admin_token'])
+            admin_token_usage[name] = privileged
```

Comment:
from reading this i understand that we are maintaining "privileged" per resource? and since many resources are within the same namespace it means we will do this multiple times? shouldn't it be per namespace?

Reply:
yes the current schema change attached to this PR is managing privileged access on the namespace level but the change to the `ResourceInventory` allows per-resource permission tracking.

when i started to write this patch, my knowledge about resource management was only basic and i made the decision to handle this task per-namespace. while digging deeper, i found that the ResourceInventory was already pretty close to what is needed to track this per-resource. the door is open to also do the last step to allow per-resource declaration in the schema.

maybe in a follow up PR? how do you feel about this?

Reply:
i was actually suggesting the other way - to do it only per namespace. happy to talk about this more if you'd like, but you can also decide yourself since things will work the same either way.

Reply:
left a comment about this in e5151b6